pcdsdevices/sqr1.py

```diff
@@ -0,0 +1,141 @@
+                         name=name,
+                         kind=kind,
+                         read_attrs=read_attrs,
+                         configuration_attrs=configuration_attrs,
+                         parent=parent,
+                         **kwargs
+                         )
+        self.x._sync_setpoints = self.sync_setpoints
+        self.y._sync_setpoints = self.sync_setpoints
+        self.z._sync_setpoints = self.sync_setpoints
+        self.rx._sync_setpoints = self.sync_setpoints
+        self.ry._sync_setpoints = self.sync_setpoints
+        self.rz._sync_setpoints = self.sync_setpoints
+
+    def sync_setpoints(self):
+        print("sync_setpoint")
+        self.x.setpoint.put(self.x.readback.get())
+        self.y.setpoint.put(self.y.readback.get())
+        self.z.setpoint.put(self.z.readback.get())
+        self.rx.setpoint.put(self.rx.readback.get())
+        self.ry.setpoint.put(self.ry.readback.get())
+        self.rz.setpoint.put(self.rz.readback.get())
+
+    def multi_axis_move(self,
+                        x_sp: float = None,
+                        y_sp: float = None,
+                        z_sp: float = None,
+                        rx_sp: float = None,
+                        ry_sp: float = None,
+                        rz_sp: float = None,
+                        wait: bool = True,
+                        timeout: float = 10.0
+                        ):
+        x_sp = self.x.readback.get() if x_sp is None else x_sp
+        y_sp = self.y.readback.get() if y_sp is None else y_sp
+        z_sp = self.z.readback.get() if z_sp is None else z_sp
+        rx_sp = self.rx.readback.get() if rx_sp is None else rx_sp
+        ry_sp = self.ry.readback.get() if ry_sp is None else ry_sp
+        rz_sp = self.rz.readback.get() if rz_sp is None else rz_sp
+
+        x_status = self.x.move(x_sp,
+                               wait=False,
+                               timeout=timeout,
+                               sync_enable=False)
+        y_status = self.y.move(y_sp,
+                               wait=False,
+                               timeout=timeout,
+                               sync_enable=False)
+        z_status = self.z.move(z_sp,
+                               wait=False,
+                               timeout=timeout,
+                               sync_enable=False
+                               )
+        rx_status = self.rx.move(rx_sp,
+                                 wait=False,
+                                 timeout=timeout,
+                                 sync_enable=False)
+        ry_status = self.ry.move(ry_sp,
+                                 wait=False,
+                                 timeout=timeout,
+                                 sync_enable=False)
+        rz_status = self.rz.move(rz_sp,
+                                 wait=False,
+                                 timeout=timeout,
+                                 sync_enable=False)
```

Comment:
Does this actually work smoothly with no hiccups? We're writing to the shared actuate signal 6 times in the code here.

Reply:
these PVs are living in the same CA server, so writing set point values to them is not a heavy process

Reply:
I guess I was more curious about how the IOC behaves. The sequence here is something like:

- call x move:
  - write to the x setpoint
  - set the actuate signal
- call y move:
  - write to the y setpoint
  - set the actuate signal
... etc.

Does this encounter any problems with e.g. starting a move to x before the rz parameter is available?

Reply:
I think this is the last functional thing to consider before merging

Reply:
I tried to move three motors at the same time and it is working fine. I got your point but it doesn't seem a problem.

Reply:
Alright, if you've tested the code as-is and there hasn't been a problem it's probably OK and shouldn't hold up the PR
I'm still worried about if of course since it's an unnecessary risk